src/components/installer/installationlayout.ts

```diff
@@ -1,9 +1,25 @@
+import * as os from 'os';
 import * as path from 'path';
 
+import { fs } from '../../fs';
 import { Platform, Shell } from "../../shell";
 
+export function vsKubernetesFolder(shell: Shell): string {
+    const originalDir = path.join(shell.home(), `.vs-kubernetes`);
+
+    if (fs.existsSync(originalDir) || os.platform() !== `linux`) {
+        return originalDir;
+    }
+
+    let xdgStateHome = process.env.XDG_STATE_HOME || ``;
+    if (xdgStateHome[0] !== `/`) {
+        xdgStateHome = path.join(os.homedir(), `.local/state`);
```

Comment:
I think it would be better to keep the default path as it is now `path.join(shell.home(), '.vs-kubernetes')`. If one deletes the existing folder for some reason and then it is recreated somewhere else the user doesn't get what's going on.
Then by doing it we would only have two different cases 1) user who doesn't use XDG specification 2) users who use it... and we don't need to remember that old linux/windows/mac users uses `.vs-kubernetes` folder, new linux users use `.local/state` and people who use xdg specification may have their custom folder.

Reply:
Ah, so if I understand correctly, your point is that it would be less confusing to users if there is an explicit "opt-in" to the "XDG Base Dir Specification". So for example, we would only follow the spec if `XDG_STATE_HOME` is set explicitly?

I feel like doing it this way would be more counterintuitive for the user and we would realize less gains from attempting to use the spec in the first place. Mostly because most users expect things to go in `~/.local/state`, `~/.config`, etc. automatically. The spirit of the spec is that this should be automatic behavior, and if we have to check if `XDG_STATE_HOME` is set (which _very_ few people actually do), then most people wouldn't be affected by this benefit (and in this case i would rather the PR not be merged since its an incomplete solution). There is also the issue of potentially having to do more of these checks for every new XDG variable used (assuming consistency is valued here) - whereas currently we only ever need to do `existsSync(~/.vs-kubernetes)` once

I think your statement about the end-user confusion when deleting `~/.vs-kubernetes` and it being recreated somewhere else is definitely valid, and I believe your mention of updating the README to address this PR in general can address that. Maybe we can say somewhere in the README along the lines of "the extension will follow the XDG Base Directory Specification appropriately (defaulting to writing config and other files in `~/.local/state/vs-kubernetes`), but only if the legacy directory of `~/.vs-kubernetes` does not exist". That way, if the user searches for "config" or ".vs-kubernetes" in the docs, they know what's up right away

Reply:
I like "use the legacy directory if it exists, otherwise use the XDG directory."  Seems like a nice pragmatic solution.